### ISSUE TITLE
wrong shader preset location since 1.8.7 on Mac

### ISSUE THREAD

**Issue by nikolafeve** [libretro/RetroArch#10956]
Retroarch still continue to write the shaders presets in the wrong directory. it write the prestes in the config directory instead of the shader directory. the bug is in the 1.8.7, 1.8.8, 1.8.9 Mac versions intel GL

**Comment by hizzlekizzle** [comment]
This change was intentional, AFAIK, and the next/previous shader presets will cycle through those custom presets now,  I believe.

**Comment by nikolafeve** [comment]
I don't get it. What the purpose to choose the shader folder in preferences if retroarch write the presets in the config folder? if i refer to the 1.8.8 update articles, the bug still here on mac version. see below:

"1.8.8 restores the original behavior of the “Save Shader Preset As” option, and improves the cycling of shaders by allowing to cycle the shaders on fallback directories if the Video Shader directory does not contain any preset."

**Comment by hizzlekizzle** [comment]
I'm not sure that "original behavior" means/meant "original location". I'll readily admit that I don't use the GUI preset-saving options, so I'm not fully abreast of the current changes.

**Comment by andreycout** [comment]
Ok, it looks really broken. All "Save Shader Preset" options save to the config directory instead of shaders/presets like before. If, however, you set the default shader directory to ":\shaders" on your retroarch.cfg, Retroarch fails to load the correct presets and when using "save shader preset as", it creates preset files with relative paths that consider user supplied cfg value for shader directory, but that are placed on the config folder, which means they also don't load correctly next time you load content.

Worse yet, there's a bug if you set your shader directory to "default" and try to manually load a preset. It asks which core you want it to open with and loads as if you're loading content instead of a shader preset. Obviously the core can't handle the slangp as content and you just get a black screen. Really weird behavior.

**Comment by nikolafeve** [comment]
Thank you @andreycout to confirm all these stranges behaviors in the last OSX versions.

**Comment by andreycout** [comment]
It's actually not just on OSX. Happened on Windows for me.

How to make it work (sort of): you need to edit retroarch.cfg and set video_shader to "default" and video_shader_dir to ":\shaders". And you need to manually setup your presets again, one by one. No way around saving to the config folder I'm afraid.

**Comment by Khronikos** [comment]
Yeah, this is on Win 7 too. Really took me for a  loop lol. Lucky I was wearing my glasses. I can confirm it is saving the presets to config. I prefer them in shaders, and I don't know why this was changed really. It's just another file we have to dig through. Pointless as the way it worked before was perfect. People dig in their shaders folders. Having to go to config where shaders are now all mixed with opt and cfg files is a bit ridiculous. Please fix these bugs or changes, they are highly annoying and easier to make right than a lot of things.

And if someone wants these changes, why? Why would anyone want to go into the config folder and then go into a separate core folder and then also have to find the game shader among various other files?
